app/components/rdf-input-fields/instance-multi-selector.js

```diff
@@ -0,0 +1,143 @@
+      headers: {
+        Accept: 'application/vnd.api+json',
+      },
+    });
+
+    this.options = await this.parseResponse(response, instanceLabelProperty);
+  }
+
+  async loadProvidedValue() {
+    if (this.isValid) {
+      const matches = triplesForPath(this.storeOptions, true).values;
+      let options = await Promise.all(
+        matches.map(async (m) => {
+          return await this.uriSearch(m.value);
+        })
+      );
+      this.selected = options
+        .map((m) => m[0])
+        .filter((opt) => matches.find((m) => m.equals(opt.subject)));
+    }
+  }
+
+  @action
+  updateSelection(options) {
+    this.selected = options;
+
+    // Retrieve options in store
+    const matches = triplesForPath(this.storeOptions, true).values;
+    const matchingOptions = matches.filter((m) =>
+      this.options.find((opt) => m.equals(opt.subject))
+    );
+
+    // Cleanup old value(s) in the store
+    matchingOptions
+      .filter((m) => !options.find((opt) => m.equals(opt.subject)))
+      .forEach((m) => updateSimpleFormValue(this.storeOptions, undefined, m));
+
+    // Insert new value in the store
+    options
+      .filter((opt) => !matchingOptions.find((m) => opt.subject.equals(m)))
+      .forEach((option) =>
+        updateSimpleFormValue(this.storeOptions, option.subject)
+      );
+
+    this.hasBeenFocused = true;
+    super.updateValidations();
+  }
+
+  search = restartableTask(async (term) => {
+    await timeout(200);
+
+    const instanceLabelProperty = this.getFormProperty('instanceLabelProperty');
+    const instanceApiUrl = this.getFormProperty('instanceApiUrl');
+
+    const url = `${instanceApiUrl}?filter=${term}`;
+    const response = await fetch(url, {
+      headers: {
+        Accept: 'application/vnd.api+json',
+      },
+    });
+    return await this.parseResponse(response, instanceLabelProperty);
+  });
+
+  getFormProperty(property) {
+    const formGraph = this.args.graphs.formGraph;
+    const FORM = new Namespace('http://lblod.data.gift/vocabularies/forms/');
+
+    return this.args.formStore.match(
+      this.args.field.uri,
+      FORM(property),
+      undefined,
+      formGraph
+    )[0].object.value;
+  }
+
+  async parseResponse(response, instanceLabelProperty) {
+    if (!response.ok) {
+      let error = new Error(response.statusText);
+      error.status = response.status;
+      throw error;
+    }
+    const result = await response.json();
+    const options = result.data.map((m) => {
+      const subject = new NamedNode(m.attributes['uri']);
+      return { subject: subject, label: m.attributes[instanceLabelProperty] };
+    });
+    return options;
+  }
+
+  // TODO we want to do this with one call, something like the following unfortunately this doesn't work at this moment.
+  // const url = `${instanceApiUrl}?filter[:uri:]=uri1,uri2`;
+  // const url = `${instanceApiUrl}?filter[:or:][:uri:]=uri1&[:or:][:uri:]=uri2`;
+  async uriSearch(term) {
```

Comment:
suggestion: rename term to targetUri? term to me sounds like you're looking up by label which you shouldn't and aren't :+1: